backend/src/index.ts

```diff
@@ -48,7 +49,11 @@ async function main() {
     throw new Error(`Wrong database version - the backend requires '${DB_VERSION}'`)
   }
 
-  // const connection = await createConnection()
+  const toCon = await typeOrmConnection()
```

Comment:
This connects? You have `// await con.connect()` in that function.

Reply:
If I call await con.connect() I get the error that the connection cannot be established because the connection already exist. 
But I have copied the code with that line from doc.

Reply:
Can you use a singleton pattern here to capsule the whole TypeORM connection mechanic handling the case of double connect? Consider this a suggestion, please do your research if its the right cause of action: 
https://stackoverflow.com/questions/50656921/typeorm-create-connection-globally
https://blog.bitsrc.io/the-singleton-pattern-in-typescript-b906303fda93

Reply:
Does that mean the main function is called more than once?
I though if I put the creation connection call in the main function it will be called only once at startup.

Reply:
That is true, but it might be easier to collect the connection object where its needed by typing
```
import connection from .../connection

const con = connection.getInstance()
con.queryStuff()
con.queryMoreStuff()
```

instead of managing the connection somewhere in the main function. Or does TypeORM that for us when utilizing their models / query builders?

Reply:
typeorm use intern a connection pool and you can get a connection with getConnection.
Entities based on BaseEntity have there own direct access to connections.
For example I can call find functions direct on the entities and typeorm handle the rest.

Reply:
Awesome, then my point is obsolete. Just keep doing <3